docs/user_guide/static_assets.md

```diff
@@ -0,0 +1,128 @@
+Each can be a list of paths, _relative to your `html_static_path`_.
+They will be added to the `<head>` of your site in order.
+
+For example:
+
+```{code-block} python
+:caption: conf.py
+
+html_static_path = ["_static"]
+html_css_files = ["mycss.css"]
+html_js_files = ["mycss.css"]
+```
+
+This will cause each to be linked in your `<head>`.
+
+## Add assets in your setup function
+
+Additionally you may add assets manually, to do so, use the `app` object in [the Sphinx `setup()` function](https://www.sphinx-doc.org/en/master/extdev/appapi.html#extension-setup).
+The `app` object has two relevant methods here:
+
+[**`app.add_css_file`**](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_css_file) allows you to add CSS files directly.
+
+[**`app.add_js_file`**](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file) allows you to add JS files directly.
+
+Both of them expect you to add files **relative to the `html_static_path`**.
+
+In addition, `app.add_js_file` allows you to add _raw JavaScript_ in addition to linking files (see example below).
+For example:
+
+```{code-block} python
+:caption: conf.py
+
+html_static_path = ["_static"]
+
+def setup(app):
+  app.add_css_file("mycss.css")
+  app.add_js_file("myjs.js")
+
+  # Add raw JavaScript
+  rawjs = """
+  let a = "foo";
+  console.log(a + " bar")
+  """
+  app.add_js_file(None, body=rawjs)
+```
+
+## Use an event to add it to specific pages
+
+If you'd like to use logic to only add a script to certain pages, or to trigger different behavior depending on the page, use [a Sphinx event hook](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx-core-events).
+This involves defining a function that runs when a particular event is emitted in the Sphinx build, and using [`app.connect()`](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.connect) to connect it to your build.
+
+The event you'll likely want to use is [`html-page-context`](https://www.sphinx-doc.org/en/master/extdev/appapi.html#event-html-page-context).
+This is triggered just before the HTML for an _individual page_ is created.
+If you run `app.add_js_file` or `app.add_css_file`, it will _only be added for that page_.
+
+For example:
+
+```python
+html_static_path = ["_static"]
+def add_my_files(app, pagename, templatename, context, doctree):
+  if pagename == "dontaddonthispage":
+    return
+
+  app.add_css_file("mycss.css")
+
+def setup(app):
+  app.connect("html-page-context", add_my_files)
+```
+
+## Add it directly to the page content
+
+Finally, you can add CSS or JS directly to a page's content.
+If you're using reStructuredText or MyST Markdown, you can use the `raw` directive:
+
+```rst
```

Comment:
so for this i'm confused - you'd add it to a rst page but it knows to add it to the <head> section when compiling? is that a sphinx feature raw will always be added to the head section of the rendered html?

Reply:
ah in this case it won't be added to the `<head>`, it'll just "exist" on the output format in raw form. So the content would look like

```
<h2>Some content

<p>A paragraph

<style>
.mystyle
</style>

</p>
```

Reply:
ah thank you for the clarification. i understand!!

Reply:
```suggestion
```{code-block} rst
:caption: some_page_in_my_site.rst
```